lib/vtls/openssl.c

```diff
@@ -1316,6 +1316,7 @@ static CURLcode ossl_set_engine(struct Curl_easy *data, const char *engine)
     const char *e_id = ENGINE_get_id(e);
     if(!strcmp(engine, e_id))
       break;
+    ENGINE_free(e);
```

Comment:
The documentation for 1.1.0 says ENGINE_free is not needed: `In a similar way, some functions will automatically release the structural reference passed to it if part of the function's job is to do so. Eg. the ENGINE_get_next() and ENGINE_get_prev() functions are used for iterating across the internal ENGINE list - they will return a new structural reference to the next (or previous) ENGINE in the list or NULL if at the end (or beginning) of the list, but in either case the structural reference passed to the function is released on behalf of the caller.` I know this code is for older versions but I'd check into whether the free is actually needed here.

Reply:
I did read the older version code and convinced myself it was required, but I don't have any notes to prove it right now. Let me go back and double-check.